ONBOARDING.md

```diff
@@ -1,10 +1,26 @@
-## So you want to write Complement tests
+# So you want to write Complement tests
 
-If you've not run Complement tests yet, please do so. This document will outline how Complement works and how you can add efficient tests and best practices for Complement itself.
+Complement is a black box integration testing framework for Matrix homeservers.
 
-### Architecture
+This document will outline how Complement works and how you can add efficient tests and best practices for Complement itself.  If you haven't run Complement tests yet, please see the [README](README.md) and start there!
+
+## Terminology
+
+* `Blueprint`: a human-readable outline for what should be done prior to a test (such as creating users, rooms, etc).
+* `Deployment`: controls the lifetime of a Docker container (built from a `Blueprint`). It has functions on it for creating deployment-scoped structs such as Client-Server API clients for interacting with specific Homeservers in the deployment.
+
+## Architecture
+
+Each Complement test runs one or more Docker containers for the homeserver(s) involved in the test. These containers are snapshots of the target homeserver at a particular state. The state of each container is determined by the `Blueprint` used. Client-Server and Server-Server API calls can then be made with assertions against the results.
+
+In order to actually write a test, Complement needs to:
+
+* Create `Deployments`, this is done by calling `deployment := Deploy(...)` (and can subsequently be killed via `deployment.Destroy(...)`).
+* (Potentially) make API calls against the `Deployments`.
+* Make assertions, which are done via the `must` and `match` packages provided by Complement.
```

Comment:
Assertions are also often made by checking a condition and then calling `t.Fatalf(msg)`, though I note the original doc didn't mention this either. @Kegsay are we trying to push people towards calling `must` and `match` instead of direct `testing` failure functions?

Reply:
Whatever is most legible. Generally `must` and `match` work well for verifying JSON objects and HTTP responses. I wouldn't be too prescriptive about it though. There's nothing wrong with `t.Fatalf`.

Reply:
Ah fair enough. I'll expand this a little!